operator/controllers/keptnworkloadinstance/reconcile_deploymentstate.go

```diff
@@ -42,24 +40,32 @@ func (r *KeptnWorkloadInstanceReconciler) reconcileDeployment(ctx context.Contex
 	return workloadInstance.Status.DeploymentStatus, nil
 }
 
-func (r *KeptnWorkloadInstanceReconciler) isReplicaSetRunning(ctx context.Context, resource klcv1alpha1.ResourceReference, namespace string) (bool, error) {
-	replica := &appsv1.ReplicaSetList{}
-	if err := r.Client.List(ctx, replica, client.InNamespace(namespace)); err != nil {
-		return false, err
-	}
-	for _, re := range replica.Items {
-		if re.UID == resource.UID {
-			replicas, err := r.getDesiredReplicas(ctx, re.OwnerReferences[0], namespace)
-			if err != nil {
-				return false, err
-			}
-			if re.Status.ReadyReplicas == replicas {
-				return true, nil
-			}
-			return false, nil
+func (r *KeptnWorkloadInstanceReconciler) isOwnerRunning(ctx context.Context, resource klcv1alpha1.ResourceReference, namespace string) (bool, error) {
```

Comment:
Should we rename this function to something like `isReferencedWorkloadRunning`? Otherwise this might be confusing as the referenced object is not the actual owner of the WorkloadInstance